bspump/main.py

```diff
@@ -1,98 +1,127 @@
+    return False
+
+
+class NotebookParser:
+    _in_autopipeline = False
+    _cell_number = 0
+    _cell_processor_contents: dict[int, str] = {}
+
+    @classmethod
+    def parse_cell(cls, cell, fout):
+        def indent_code(lines: list[str]) -> list[str]:
+            multiline = False
+            double_quotes = False
+            indent_lines = []
+            lines_out = []
+            for i, line in enumerate(lines):
+                if not multiline and line.strip(" ") != "":
+                    indent_lines.append(i)
+                if '"""' in line or "'''" in line:
+                    if not multiline:
+                        double_quotes = '"""' in line
+                    elif (double_quotes and "'''" in line) or (
+                        not double_quotes and '"""' in line
+                    ):
+                        continue
+                    multiline = not multiline
                     continue
-                clean_code += line + "\n"
-            __bs_cell_code_contents[cell_number] = clean_code
-            try:
-                if bspump.jupyter.bitswan_auto_pipeline.get("sink") is not None:
-                    clean_code = f"""
-global __bs_step_locals
-# if undefined define __bs_step_locals as empty dict
-if not "__bs_step_locals" in globals():
-    __bs_step_locals = {{}}
-# load locals from __bs_step_locals
-__bs_step_locals['event'] = event
-exec(__bs_cell_code_contents[{cell_number}] + "__bs_step_locals = locals()\\ndel __bs_step_locals['__bs_step_locals']",globals(), __bs_step_locals)
-return __bs_step_locals['event']
-"""
-
-                    parsed_code = ast.parse(clean_code)
-
-                    # Step 3: Create a new function definition
-                    new_function = ast.FunctionDef(
-                        name=f"step_{cell_number}_internal",
-                        args=ast.arguments(
-                            posonlyargs=[],
-                            args=[ast.arg(arg="event", annotation=None)],
-                            kwonlyargs=[],
-                            kw_defaults=[],
-                            defaults=[],
-                        ),
-                        body=parsed_code.body,
-                        decorator_list=[ast.Name(id="step", ctx=ast.Load())],
-                    )
-
-                    # Step 4: Wrap the function definition in a module
-                    module = ast.Module(body=[new_function], type_ignores=[])
-
-                    # Step 5: Compile the module
-                    module = ast.fix_missing_locations(module)
-                    compiled_code = compile(module, filename="<ast>", mode="exec")
-                    # exec the code
-                    exec(compiled_code, ctx)
+            for i in range(len(lines)):
+                _indent = "    " if i in indent_lines else ""
+                lines_out.append(_indent + lines[i])
+            return lines_out
+
+        def cell_str(contents, indent=False):
```

Comment:
This function is unnecessary and just makes the code more confusing, inline this logic.